PORTING.md

```diff
@@ -0,0 +1,126 @@
+string constants.
+
+Alternative operator representations such as 'not', 'and', 'compl' etc. which are alternatives
+to '!', '&&', '~' etc. which had been ISO standard workarounds for very limited character sets and
+locale-specific keyboards are still supported by gcc and clang, but were never supported by the 
+Microsoft compiler and are now deprecated by the language spec.
+
+Whenever platform-specific header files are needed, include the envoy/common/platform.h file to pick
+up the most common inclusions. This file makes a number of adjustments to define common constructs
+that may be missing on a particular environment. This file also adjusts for system file misbehavior.
+(For example, the Windows.h header file #define's many common words that break even the simplest C++
+source code; platform.h reverts some of this damage).
+
+The portable implementations of FilesystemImpl, IoSocketImpl and low-level OsSysCallsImpl using
+os_fd_t descriptors and the error constants defined in `platform.h` should be used in favor of raw
+POSIX APIs when authoring tests.
+
+## Dependency behavioral differences
+
+The libraries that Envoy depends on will exhibit different behaviors when built for Windows
+versus Linux versus BSD architectures. In some cases, Envoy has patched the library to either
+correct misbehavior, or to make the library follow expected linux conventions more closely,
+and every attempt is made to push such changes as optional features to upstream. Some notable
+discrepancies include;
+
+The googletest library has a number of Windows-specific quirks. At the time of this writing,
+this library is incompatible with the LLVM clang-cl. It uses the POSIX re (regular expression)
+system library by default. There is an option to use the PCRE library instead, which the
+project has not adopted. On Windows, it uses a massively simplified expression parser which
+is missing support for most common Re expressions. (The solution to this, substituting the
+Google Re library, is proposed but has not been adopted by the googletest project yet).
+
+The opentracing library is missing a Windows build schema (it relies entirely on automake),
+so all tracing-related Envoy extensions are disabled on Windows.
+
+## Bazel behavior differences
+
+There are several circumstances that bazel will end up misbehaving in one environment or another.
+For example, during RBE CI builds, the environment, paths, mounts etc. may have longer paths or
+otherwise hit limits that aren't observed on the contributor's environment.
+
+Similarly, when building on Windows the tools themselves have different path or command line length
+restrictions than on the typical Linux/clang tool chain. These limits include;
+
+* The individual path elements for -Inclusion (this restricts the total path name lengths available to the project).
+* The length of the embedded command line stored in the debug info record of the COFF object file (48kb on Windows).
```

Comment:
How can developers action on this to prevent running into this situation?

Reply:
We haven't worked out that solution yet. It's noted here in case Windows developers are observing strange debugging behavior, it isn't something we expect to see in CI. The way that Bazel assembles works with insane lists of inclusions suggests we won't be able to work around this. We can probably pull the comment if we don't see issues in CI or from developers, since there isn't anything that Envoy itself can realistically do.

Reply:
Note to self to inject bazel advice to Windows builders here, as mentioned in https://github.com/envoyproxy/envoy/issues/12098